apps/etterlatte-behandling/src/main/kotlin/oppgave/OppgaveService.kt

```diff
@@ -228,53 +224,85 @@ class OppgaveService(
         }
     }
 
-    fun ferdigstillOppgaveUnderbehandlingOgLagNyMedType(
-        fattetoppgaveReferanseOgSak: SakIdOgReferanse,
-        oppgaveType: OppgaveType,
+    fun tilAttestering(
+        referanse: String,
+        type: OppgaveType,
         merknad: String?,
-        saksbehandler: BrukerTokenInfo,
+        frist: Tidspunkt? = null,
     ): OppgaveIntern {
-        val behandlingsoppgaver = oppgaveDao.hentOppgaverForReferanse(fattetoppgaveReferanseOgSak.referanse)
-        if (behandlingsoppgaver.isEmpty()) {
-            throw BadRequestException("Må ha en oppgave for å kunne lage attesteringsoppgave")
+        val oppgaver =
+            hentOppgaverForReferanse(referanse)
+                .filter { it.type == type }
+                .filter { it.status in listOf(Status.UNDER_BEHANDLING, Status.UNDERKJENT) }
+
+        if (oppgaver.isEmpty()) {
+            throw ManglerOppgaveUnderBehandling("Ingen oppgave funnet for referanse: $referanse")
+        } else if (oppgaver.size > 1) {
+            throw ForMangeOppgaverUnderBehandling("For mange oppgaver under behandling (ref: $referanse)")
         }
-        try {
-            val oppgaveUnderbehandling = behandlingsoppgaver.single { it.status == Status.UNDER_BEHANDLING }
-            ferdigstillOppgaveById(oppgaveUnderbehandling, saksbehandler)
-            return opprettNyOppgaveMedSakOgReferanse(
-                referanse = fattetoppgaveReferanseOgSak.referanse,
-                sakId = fattetoppgaveReferanseOgSak.sakId,
-                oppgaveKilde = oppgaveUnderbehandling.kilde,
-                oppgaveType = oppgaveType,
-                merknad = merknad,
-            )
-        } catch (e: NoSuchElementException) {
-            throw BadRequestException(
-                "Det må finnes en oppgave under behandling, gjelder behandling:" +
-                    " ${fattetoppgaveReferanseOgSak.referanse}",
-                e,
-            )
-        } catch (e: IllegalArgumentException) {
-            throw BadRequestException(
-                "Skal kun ha en oppgave under behandling, gjelder behandling:" +
-                    " ${fattetoppgaveReferanseOgSak.referanse}",
-                e,
-            )
+
+        val oppgave = oppgaver.single()
+
+        val oppdatertMerknad = merknad ?: oppgave.merknad ?: ""
+        oppgaveDao.oppdaterStatusOgMerknad(oppgave.id, oppdatertMerknad, Status.ATTESTERING)
```

Comment:
Du har det sikkert på blokka men bare nevner det just in case, at vi kanskje kan slå sammen noen av dao oppdateringene her.

Reply:
Ja, tenker det er flere steder som kan skrives om litt for å gjøre det enklere, men vil ikke ta det i denne PR-en. Greit å ikke ta for mye på én gang.